File: Python/algorithms/sorting/bubble_sort.py
from typing import List


def swap(list: List[int], a: int, b: int) -> None:
    list[a], list[b] = list[b], list[a]


def bubble_sort(list: List[int]) -> None:
    i = 0
    while i < len(list):
        j = len(list) - 1
        while j > i:
            if list[j] < list[j - 1]:
                swap(list, j, j-1)
            j -= 1
        i += 1


list = [29, 99, 27, 41, 66, 28, 44, 78, 87,
        19, 31, 76, 58, 88, 83, 97, 12, 21, 44]
print(list)
bubble_sort(list)
print(list)

File: TypeScript/src/algorithms/sorting/bubbleSort.ts
import { swap } from './../../util'

export const bubbleSort = (array: number[]) => {
  const sortedArray = array.slice()

  for (let i = 0; i < sortedArray.length - 1; i++) {
    for (let j = sortedArray.length - 1; j > i; j--) {
      if (sortedArray[j] < sortedArray[j - 1]) {
        swap(sortedArray, j, j - 1)
      }
    }
  }

  return sortedArray
}
